api/src/paths/project/{projectId}/survey/{surveyId}/template/{templateId}/getSignedUrl.ts

```diff
@@ -77,11 +77,25 @@ export function getSingleTemplateURL(): RequestHandler {
         getSurveyTemplateS3KeySQLStatement.values
       );
 
+      console.log('result is: ', result);
+
       await connection.commit();
 
-      const s3Key = result && result.rows.length && result.rows[0].key;
+      let s3SignedUrl;
+
+      if (result && result.rows.length) {
+        const originalName = result.rows[0].file_name;
+        const templateId = result.rows[0].occurrence_submission_id;
 
-      const s3SignedUrl = await getS3SignedURL(s3Key);
+        const s3Key = generateS3FileKey({
+          projectId: Number(req.params.projectId),
+          surveyId: Number(req.params.surveyId),
+          folder: `submissions/${templateId}`,
+          fileName: originalName
+        });
```

Comment:
We no longer need to generate the key, now that we store the full key in the database.